frontend/src/metabase-lib/lib/metadata/Field.ts

```diff
@@ -43,12 +44,13 @@ import Base from "./Base";
  */
 
 class FieldInner extends Base {
-  id?: number;
+  id: number | FieldRef;
```

Comment:
See the `ref` method -- this is apparently a possibility. I think in most scenarios you'll see a `number` id, but I guess it happens somewhere. Probably shouldn't be a `Field` in that scenario, but it is.